### PR TITLE
Update derive_vars_dtm to change min_dates with max_dates for lapply …

### PR DESCRIPTION
**Pull request by hamzarahal**
Update derive_vars_dtm to put max_dates instead of min_dates when calling lapply
Update the NEWS to describe the bug
Add unit test to cover the case

### CODE REVIEW

File: tests/testthat/test-derive_vars_dtm.R [hunk 1]
```diff
@@ -203,3 +203,31 @@ test_that("No re-derivation is done if --DTF variable already exists", {
...
+    date_imputation = "MID"
+  )
+
+
+  expect_equal(
+    expected_output,
+    actual_output
+  )
+
+})
```
Comment (thomas-neitmann) [inline tests/testthat/test-derive_vars_dtm.R:233]: This should be a test that makes sure the `max_dates` work as expected.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

File: R/derive_date_vars.R [hunk 2]
```diff
@@ -699,7 +699,7 @@ derive_vars_dtm <- function(dataset,
     date_imputation = date_imputation,
     time_imputation = time_imputation,
     min_dates = lapply(min_dates, eval_tidy, data = mask),
-    max_dates = lapply(min_dates, eval_tidy, data = mask)
+    max_dates = lapply(max_dates, eval_tidy, data = mask)
```
Comment (thomas-neitmann) [inline R/derive_date_vars.R:702]: Can you please check if the same error is made in `derive_vars_dt()`?

File: tests/testthat/test-derive_vars_dtm.R (hunk 1 above)
Comment (hamzarahal) [reply tests/testthat/test-derive_vars_dtm.R:233]: maybe we need to call max_dates and min_dates in such test

File: R/derive_date_vars.R (hunk 2 above)
Comment (hamzarahal) [reply R/derive_date_vars.R:702]: it looks fine for derive_vars_dt()

File: tests/testthat/test-derive_vars_dtm.R (hunk 1 above)
Comment (thomas-neitmann) [reply tests/testthat/test-derive_vars_dtm.R:233]: Yes, exactly.

Comment (hamzarahal) [reply tests/testthat/test-derive_vars_dtm.R]: Th is the truncated result comparing with the content of test_that() message as dicussed yesterday, but I guess no impact on test result as we get the expected warning

image

**Comment by thomas-neitmann** [comment]
@traceyxw @makropoa @wqrena Could one of you pick up this PR? What's missing is a unit test.

**Comment by amsmith214** [comment]
@wqrena sorry-- i think i got caught up in this by trying to fix #698, which had some work this end. can you check the unit test i've added there? i added you as a reviewer for that one. i think that PR #762 might actually fix the template documentation issue and this issue

**Comment by wqrena** [comment]
@thomas-neitmann, I added unit test for max_date and min_date. It should be able to detect max_date bugs in #687 before PR.

**Comment by wqrena** [comment]
@amsmith214, just realized that the PR #762 did the unit test for max_date as well. I reviewed it, the unit test works well in PR#762. In this PR#702, the unit test had the unit test done for both max_date and min_date. I guess we may need to keep one unit test.

**Comment by amsmith214** [comment]
@pharmaverse/core-team i think #762 fixes this-- if that one merges we could close this

**Comment by thomas-neitmann** [comment]
Closing this one in favor of #762. In the future please avoid putting fixes for multiple issues into one PR.

**Comment by amsmith214** [comment]
> Closing this one in favor of #762. In the future please avoid putting fixes for multiple issues into one PR.

understood; apologies!

**Pull request closed without merging**